src/material.js

```diff
@@ -319,4 +319,44 @@ export class Material extends HasMetadataNamedDefaultableInMemoryEntity {
 
         return new this.constructor(config);
     }
+
+    /**
+     * @summary a series of checks for the material and returns an array of results in ConsistencyChecks format.
+     * @returns {{keys: *[], messages: *[]}} - Array of checks results
+     */
+    getConsistencyChecks() {
+        const basisChecks = this.getBasisConsistencyChecks();
+
+        // any other Material checks can be added here
+
+        return {
+            keys: Array.from(basisChecks.keys),
+            messages: basisChecks.messages,
+        };
+    }
+
+    /**
+     * @summary a series of checks for the material's basis and returns an array of results in ConsistencyChecks format.
+     * @returns {{keys: Set<any>, messages: *[]}}
+     */
+    getBasisConsistencyChecks() {
+        const keys = new Set();
+        const messages = [];
+
+        const basis = this.Basis;
+        const overlappingAtomsGroups = basis.getOverlappingAtoms();
+
+        overlappingAtomsGroups.forEach(({ id1, id2, element1, element2 }) => {
+            const key = `basis.coordinates.${id1}`;
+            keys.add(key);
+            messages.push({
+                key,
+                checkName: "atomsOverlap",
```

Comment:
Why do we hardcode "atomsOverlap"?

Reply:
Ideally, it should be something like `checkNamesEnum.material.basis.atomsOverlap` stored in some centralized repository. This sounds like definitions, core to the entities that get checked -- code.js ?
But for now not to overoptimize I left it like that.